storage/postgres/query_builder.go

```diff
@@ -12,23 +15,74 @@ import (
 	"github.com/jmoiron/sqlx"
 )
 
-const mainTableAlias = "t"
-
-type orderRule struct {
-	field     string
-	orderType query.OrderType
-}
-
-type queryStringBuilder struct {
-	strings.Builder
-}
-
-func (qsb *queryStringBuilder) Replace(old, new string) {
-	current := qsb.String()
-	qsb.Reset()
-	current = strings.Replace(current, old, new, 1)
-	qsb.WriteString(current)
-}
+const PrimaryKeyColumn = "id"
+
+const COUNTWithoutCriteriaTemplate = `
+SELECT COUNT(DISTINCT {{.ENTITY_TABLE}}.{{.PRIMARY_KEY}})
+FROM {{.ENTITY_TABLE}}
+{{.FOR_SHARE_OF}}
+{{.LIMIT}};`
```

Comment:
Try to remove the limit from count queries by calculating a specific set of data for template resolving per operation instead of one with all the clauses for each operation.